pkg/apis/scheduling/types.go

```diff
@@ -79,3 +82,42 @@ type PriorityClassList struct {
 	// Items is the list of PriorityClasses.
 	Items []PriorityClass
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PodGroup is a collection of Pod; used for batch workload.
+type PodGroup struct {
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+
+	// Spec defines the behavior of a pod group.
+	// +optional
+	Spec PodGroupSpec
+}
+
+// PodGroupSpec is a description of a pod group.
+type PodGroupSpec struct {
+	// NumMember defines the number of members/tasks to run the pod group;
+	// if there's not enough resources to start all tasks, the scheduler
+	// will not start anyone.
+	NumMember int32
+	// Resources defines the total resource the PodGroup requests to run
+	// Pods.
+	Resources map[string]resource.Quantity
```

Comment:
@liggitt  ,  do you know how to use `v1.ResourceList` here? The `make update` did not include `k8s.io/apimachinery/pkg/api/resource` in `generated.pb.go`.

```bash
vendor/k8s.io/api/scheduling/v1alpha1/generated.pb.go:850:18: undefined: k8s_io_apimachinery_pkg_api_resource
vendor/k8s.io/api/scheduling/v1alpha1/generated.pb.go:857:18: undefined: k8s_io_apimachinery_pkg_api_resource
```